### PR TITLE
Auto login

### PR DESCRIPTION
**Pull request by davidthomas22**
### Context

> Does this issue have a Trello card?

> If this is an issue, do we have steps to reproduce?

### Intent

> What changes are introduced by this PR that correspond to the above card?

> Would this PR benefit from screenshots?

### Considerations

> Is there any additional information that would help when reviewing this PR?

> Are there any steps required when merging/deploying this PR?

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: server/auth/middleware.js [hunk 1]
```diff
@@ -141,9 +152,10 @@ const createSignInCallbackMiddleware = ({
 
 const createSignOutMiddleware = ({ logger, analyticsService }) =>
   function signOut(req, res) {
-    if (req.user?.isSignedIn()) {
+    if (req.session?.isSignedIn) {
       logger.info(`SignOutMiddleware (signOut) - User: ${req.user.prisonerId}`);
-      req.logOut();
+      req.session.isSignedIn = false;
+      // req.logOut();
```
Comment (jonhindle) [inline server/auth/middleware.js:158]: Commented out for now. I think we should leave users logged in though.

File: server/__tests__/app.spec.js [hunk 2]
```diff
@@ -88,18 +106,32 @@ describe('Sentry', () => {
...
+      .then(({ text }) => {
+        expect(text).toContain('Found. Redirecting to ');
+      });
+  });
   it('renders a 404 page correctly on invalid url', async () => {
     config.auth.callbackPath = '/testPath';
     await request(app())
       .get('/unknown-url')
-      .expect(404)
+      .set('host', 'wayland.prisoner-content-hub')
```
Comment (andrewrlee) [inline server/__tests__/app.spec.js:134]: oh cool, didn't know you could do this

File: server/app.js [hunk 3]
```diff
@@ -172,12 +171,12 @@ const createApp = services => {
...
   app.use(featureToggleMiddleware(config.features));
 
-  // establishment toggle
-  app.use(configureEstablishment());
-
   // Health end point
   app.use('/health', createHealthRouter(config));
 
+  // establishment toggle
+  app.use(getEstablishmentFromUrl);
```
Comment (andrewrlee) [inline server/app.js:178]: nice rename - I was going to suggest but totally forgot to

File: server/auth/middleware.js [hunk 4]
```diff
@@ -108,8 +115,12 @@ const createSignInCallbackMiddleware = ({
         const { bookingId, agencyId } =
           await offenderService.getOffenderDetailsFor(user);
         user.setBookingId(bookingId);
-
-        updateSessionEstablishment(req, agencyId);
+        if (!req.session?.establishmentName) {
+          updateSessionEstablishment(req, agencyId);
+          req.session.isSignedIn = false;
+        } else {
+          req.session.isSignedIn = true;
```
Comment (andrewrlee) [inline server/auth/middleware.js:122]: mmh.. why don't we need to always `updateSessionEstablishment`?

File: server/middleware/__tests__/getEstablishmentFromUrl.spec.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  it('should retrieve establishment from the header and set in the session', () => {
+    getEstablishmentFromUrl(req, res, next);
+    expect(req.session.establishmentName).toBe('cookhamwood');
+  });
+  it('should not retrieve establishment from the header and set in the session if not provided', () => {
+    req.headers.host = urlStub;
+    getEstablishmentFromUrl(req, res, next);
+    expect(req.session.establishmentName).not.toBeDefined();
+  });
```
Comment (andrewrlee) [inline server/middleware/__tests__/getEstablishmentFromUrl.spec.js:26]: might be good to have tests for all the possible versions of the url?:

localhost:3000
wayland.prisoner-content-hub.local:3000
wayland-prisoner-content-hub-development-123.apps.live-1.cloud-platform.service.justice.gov.uk
wayland-prisoner-content-hub-staging.apps.live-1.cloud-platform.service.justice.gov.uk/
wayland-prisoner-content-hub-staging.apps.live-1.cloud-platform.service.justice.gov.uk/
content-hub.prisoner.service.justice.gov.uk
wayland.content-hub.prisoner.service.justice.gov.uk

File: helm_deploy/prisoner-content-hub-frontend/values-development.yaml [hunk 6]
```diff
@@ -7,7 +7,7 @@ application:
     feedbackServiceName: aws-es-proxy-service
     feedbackEndpoint: /prod-feedback/_doc
     analyticsSiteId: UA-152065860-4
-    mockAuthEnabled: true
+    mockAuthEnabled: false
```
Comment (andrewrlee) [inline helm_deploy/prisoner-content-hub-frontend/values-development.yaml:10]: is the plan to include mockAuthEnabled false in dev for now?

obviously you needed this to test but do you think we should have this as the default for a bit?

It seems like mockAuthEnabled: true is helpful when working on profile bits which we haven't done for a while

(I think we should fix the urls so we can support wildcards so it's less of an issue switching between the two modes - but outside of this ticket)

File: server/app.js [hunk 7]
```diff
@@ -225,6 +224,23 @@ const createApp = services => {
     );
   }
 
+  app.use((req, res, next) => {
+    logger.debug(
+      `*********** establishment: ${req.session?.establishmentName} user: ${req.user?.prisonerId}`,
+    );
```
Comment (andrewrlee) [inline server/app.js:230]: presumably we're ditching these before merging?

File: server/auth/middleware.js (hunk 1 above)
Comment (andrewrlee) [reply server/auth/middleware.js:164]: happy to go with that

File: helm_deploy/prisoner-content-hub-frontend/values-development.yaml (hunk 6 above)
Comment (davidthomas22) [reply helm_deploy/prisoner-content-hub-frontend/values-development.yaml:10]: Agreed, I think pushing the fix urls up the backlog would be a good idea

**Review by andrewrlee — APPROVED** [review]
LGTM

File: server/auth/middleware.js [hunk 8]
```diff
@@ -63,11 +69,21 @@ const createMockSignIn = ({ offenderService }) =>
...
+      } else {
+        req.session.isSignedIn = true;
+      }
 
       req.session.passport = {
         user: user.serialize(),
       };
+      logger?.debug(`>>>>>>>>>> user: ${user.prisonerId}`);
+      logger?.debug(`>>>>>>>>>> signed in: ${req.session?.isSignedIn}`);
+      logger?.debug(
```
Comment (andrewrlee) [inline server/auth/middleware.js:84]: logging?

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**